### PR TITLE
Prepare Release 6.1.0

### PR DESCRIPTION
**Pull request by jacekwegr**
## Highlights

- Reworked C2S architecture
- Docker image for arm64
- Update of Universal Analytics to Google Analytics 4
- Dialyzer types fix
- Deprecation of Riak
- More reliable test suites
- Various improvements and fixes

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174498) / small_tests / b74839c3eff7a98eae5708805eb59d50b844b16f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174498/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174498/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-04_13.36.54/lib.mongooseim.logs/run.2023-05-04_13.36.54/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174526) / small_tests / 473eef739e4be161c218ff1b32fdbdc2b910ff8a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174526/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174526/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-04_13.40.55/lib.mongooseim.logs/run.2023-05-04_13.40.56/suite.log.html)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,99 @@
...
+  - `ping` (#3748)
+  - `metrics` (#3800)
+  - `steam_management` (#3796)
+  - `mod_csi` (#3880)
+  - `sasl_external` (#3911)
+  - More granular hooks (#3852, #3857, #3955)
+  - Fixed `stream_management` timeouts (#3934)
+  - Stopped routing broadcast tuples (#3946)
+  - Unified metrics (#3967)
+  - Migration guide (#3965)
```
Comment (JanuszJakubiec) [inline CHANGELOG.md:30]:
```suggestion
  - C2S Migration guide (#3965)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,99 @@
+# [MongooseIM 6.1.0](https://github.com/esl/MongooseIM/releases/tag/6.1.0) - 2023-05-10
+
+## Highlights
+
+- Reworked C2S architecture
+- Docker image for arm64
+- Better Dialyzer
```
Comment (JanuszJakubiec) [inline CHANGELOG.md:7]:
```suggestion
- Dialyzer types fix
```

Comment (JanuszJakubiec) [reply CHANGELOG.md:7]: It's just a suggestion, for me the "Better Dialyzer" is a bit misleading.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,99 @@
+# [MongooseIM 6.1.0](https://github.com/esl/MongooseIM/releases/tag/6.1.0) - 2023-05-10
+
+## Highlights
```
Comment (JanuszJakubiec) [inline CHANGELOG.md:3]: I think the update of Universal Analytics to Google Analytics 4 deserves the inclusion to highlights, but I'm not 100% sure.

**Review by JanuszJakubiec — APPROVED** [review]
I've added two comments. They are more a suggestion than change requests. Overall it looks good.

File: CHANGELOG.md (hunk 3 above)
Comment (jacekwegr) [reply CHANGELOG.md:3]: Yeah, you're right. I wrote the highlights section before the merge of the PR with GA4 and forgot to update it.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,100 @@
...
+  - `metrics` (#3800)
+  - `steam_management` (#3796)
+  - `mod_csi` (#3880)
+  - `sasl_external` (#3911)
+  - More granular hooks (#3852, #3857, #3955)
+  - Fixed `stream_management` timeouts (#3934)
+  - Stopped routing broadcast tuples (#3946)
+  - Unified metrics (#3967)
+  - C2S Migration guide (#3965)
+  - Fixed duplicated logout in `steam_management` (#3983)
```
Comment (chrzaszcz) [inline CHANGELOG.md:32]: 😉 
```suggestion
  - Fixed duplicated logout in `stream_management` (#3983)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,100 @@
...
+  - Adapted existing tests suites (#3772, #3778, #3783, #3787, #3786,  #3845, #3882, #3909, #3925, #3927, #3932, #3931, #3940)
```
Comment (chrzaszcz) [inline CHANGELOG.md:33]:
```suggestion
  - Adapted existing test suites (#3772, #3778, #3783, #3787, #3786,  #3845, #3882, #3909, #3925, #3927, #3932, #3931, #3940)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,100 @@
...
+- Checking for MUC domain when archiving messages (#3936)
+- Updated base Docker image (#3943)
+- Implemented `mod_pubsub_db` without dynamic modules (#3953)
+- Checking the `to` JID (#3971)
+- Improved pagination in `inbox` (#3974)
+- Auto registration/unregistration of hooks in `gen_mod` (#3954)
+- Changed version of eodbc to 0.2.0 (#3976)
+- Added port and IP in the listener options for WebSockets and BOSH (#3977)
+- Improved MAM logging (#3984)
+- Reason to batch worker termination (#3985)
```
Comment (chrzaszcz) [inline CHANGELOG.md:49]:
```suggestion
- Added error reason to batch worker termination (#3985)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,100 @@
...
+  - Stopped routing broadcast tuples (#3946)
+  - Unified metrics (#3967)
+  - C2S Migration guide (#3965)
+  - Fixed duplicated logout in `steam_management` (#3983)
+  - Adapted existing tests suites (#3772, #3778, #3783, #3787, #3786,  #3845, #3882, #3909, #3925, #3927, #3932, #3931, #3940)
+  - Miscellaneous changes (#3729, #3797, #3790, #3816, #3858, #3888, #3904, #3908, #3917, #3919, #3935, #3950, #3957, #3959, #3972)
+- Added information about GraphQL (#3905)
+- Enable CircleCI tests insights (#3899)
+- Hooks improvement (#3913, #3912)
+- `inbox` improvement (#3910)
```
Comment (chrzaszcz) [inline CHANGELOG.md:38]: It might be good to group inbox extensions under one item - just grep for `inbox` in this list.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,100 @@
...
+- Implemented `mod_pubsub_db` without dynamic modules (#3953)
+- Checking the `to` JID (#3971)
+- Improved pagination in `inbox` (#3974)
+- Auto registration/unregistration of hooks in `gen_mod` (#3954)
+- Changed version of eodbc to 0.2.0 (#3976)
+- Added port and IP in the listener options for WebSockets and BOSH (#3977)
+- Improved MAM logging (#3984)
+- Reason to batch worker termination (#3985)
+- Docker image for arm64 (#3979, #3986, #3988)
+- tlsv1.3 (#3989)
```
Comment (chrzaszcz) [inline CHANGELOG.md:51]: Full support for TLS version 1.3

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,100 @@
...
+- Status badges (#3975)
+- Announcing session establishment (#3970)
+- Input for `mam` errors (#4008)
+- Error 404 when getting dependencies on Debian (#4017)
+- Issue with certificate verification for Google Analytics (#3978)
+- Incremental removal query (#3924)
+- Unknown types in Dialyzer (#3929)
+
+### Removed
+- Ejabberd from C2S (#3805, #3860)
```
Comment (chrzaszcz) [inline CHANGELOG.md:80]:
```suggestion
- Legacy C2S implementation (#3805, #3860)
```

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general. Please think if it is possible to group the Docker ARM64 PR's in the change log.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174642) / small_tests / 6b61ad97e18c4848147f33c651409eb3098921fa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174642/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174642/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-09_08.04.11/lib.mongooseim.logs/run.2023-05-09_08.04.11/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174767) / small_tests / 14358013f1a26e252b24093279888c8e38e922bb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174767/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174767/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-09_12.34.05/lib.mongooseim.logs/run.2023-05-09_12.34.06/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174846) / small_tests / 53d2cda4c2ffb6391d88deb5b3cf517a39707006
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174846/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174846/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-09_14.45.13/lib.mongooseim.logs/run.2023-05-09_14.45.13/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/174876) / small_tests / 43f1c783ffcf78d8a1becbe3b8bd31e18c9a52a4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4018/174876/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4018/174876/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-05-09_14.48.53/lib.mongooseim.logs/run.2023-05-09_14.48.54/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good, we are all set for the release 👍

**Pull request merged**